src/v/raft/consensus.cc

```diff
@@ -722,30 +721,38 @@ consensus::linearizable_barrier(model::timeout_clock::time_point deadline) {
         update_node_append_timestamp(target);
         vlog(
           _ctxlog.trace, "Sending empty append entries request to {}", target);
-        auto f = _client_protocol
-                   .append_entries(
-                     target.id(),
-                     std::move(req),
-                     rpc::client_opts(_replicate_append_timeout))
-                   .then([this, id = target.id(), seq, dirty_offset](
-                           result<append_entries_reply> reply) {
-                       process_append_entries_reply(
-                         id, reply, seq, dirty_offset);
-                   });
-
-        send_futures.push_back(std::move(f));
+
+        ssx::spawn_with_gate(
+          _bg,
+          [this,
+           id = target.id(),
+           req = std::move(req),
+           seq,
+           dirty_offset]() mutable {
+              return _client_protocol
+                .append_entries(
+                  id,
+                  std::move(req),
+                  rpc::client_opts(_replicate_append_timeout))
+                .then([this, id, seq, dirty_offset](
+                        result<append_entries_reply> reply) {
+                    process_append_entries_reply(id, reply, seq, dirty_offset);
+                })
+                .handle_exception([this, id](const std::exception_ptr& e) {
+                    vlog(
+                      _ctxlog.warn,
+                      "append entries dispatch to node {} failed: {}",
+                      id,
+                      e);
+                });
+          });
     });
     // release semaphore
     // term snapshot taken under the semaphore
     auto term = _term;
 
     u.return_all();
 
-    // wait for responses in background
-    ssx::spawn_with_gate(_bg, [futures = std::move(send_futures)]() mutable {
-        return ss::when_all_succeed(futures.begin(), futures.end());
```

Comment:
where was the exceptional future happening? this should wait until all the futures are ready, at which point you could attach a continuation to when_all_succeed that ignores the exceptional ones or does whatever you want with them.

Reply:
The issue is gate may be closed by this point (so no when_all..)  and the futures are destroyed and maybe exceptional. The rewrite handles this case without having to when_all.. and rather rely on the same gate.

Reply:
ah got it thx